qutebrowser/utils/urlutils.py

```diff
@@ -471,24 +474,46 @@ def incdec_number(url, incdec):
     if not url.isValid():
         raise InvalidUrlError(url)
 
-    path = url.path()
-    # Get the last number in a string
-    match = re.match(r'(.*\D|^)(\d+)(.*)', path)
-    if not match:
-        raise IncDecError("No number found in URL!", url)
-    pre, number, post = match.groups()
-    # This should always succeed because we match \d+
-    val = int(number)
-    if incdec == 'decrement':
-        if val <= 0:
-            raise IncDecError("Can't decrement {}!".format(val), url)
-        val -= 1
-    elif incdec == 'increment':
-        val += 1
-    else:
-        raise ValueError("Invalid value {} for indec!".format(incdec))
-    new_path = ''.join([pre, str(val), post])
+    if segments is None:
+        segments = {'path', 'query'}
+    valid_segments = {'host', 'path', 'query', 'anchor'}
+    if segments - valid_segments:
+        extra_elements = segments - valid_segments
+        raise IncDecError("Invalid segments: {}".format(
+            ', '.join(extra_elements)), url)
```

Comment:
Nothing should happen, those "segments" would be just ignored, so it's not particularly bad, but might lead to bugs (e.g. passing "domain" instead of "host", which would be silently accepted).

I duplicated the check because the function is part of `urlutils` and thus not bound to being called by `_navigate_incdec`, so the values may not always come from the config. If you deem the double-check unnecessary, I will remove it.